File: java/face/src/main/java/co/edu/eafit/BusquedasIncrementales.java
package co.edu.eafit;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


public class BusquedasIncrementales {
    private final Evaluator eval;
    private final BufferedReader br;


    public BusquedasIncrementales() {
        eval = new Evaluator();
        br = new BufferedReader(new InputStreamReader(System.in));
    }


    private void algorithm(String fx, double x0, double delta, double niter){
        double fx0, fx1, x1, zero = 0, contador;
        
        fx0 = eval.evaluatorExpression(fx, x0);

        if(fx0 == zero){
            System.out.println(x0 + "es raiz");

        } else {
            x1 = x0 + delta;
            contador = 1;

            fx1 = eval.evaluatorExpression(fx, x1);
            while((fx0 * fx1) > zero && (contador <= niter)){
                x0 = x1;
                fx0 = fx1;
                x1 = x0 + delta;
                fx1 = eval.evaluatorExpression(fx, x1);
                contador += 1;
            }

            if(fx1 == zero){
                System.out.println(x0 + " es raiz");
            }else if ((fx0 * fx1) < 0){
                System.out.println("Hay una raiz entre " + x0 + " y " + x1);
            }else{
                System.out.println("Fracaso en " + niter + " iteraciones");
            }
        }
    }



    public void controller() throws IOException{
        
        System.out.print("f(x) = ");
        String fx = br.readLine();

        System.out.print("x0 = ");
        String x0Str = br.readLine();
        double x0 = Double.parseDouble(x0Str);

        System.out.print("Delta = ");
        String deltaStr = br.readLine();
        double delta = Double.parseDouble(deltaStr);

        System.out.print("Num. Iteraciones = ");
        String niterStr = br.readLine();
        double niter = Double.parseDouble(niterStr);

        algorithm(fx, x0, delta, niter);
    }


}

File: python/face/api/metodos/busquedasIncrementales.py
from math import e, log
from sympy import *

x = Symbol('x')


x0 = 1
delta =  1.0001
n_iter = 100
f = log(x+2) - x

fx0 = f.evalf(subs={x:x0})
if fx0 == 0:
    print("{0} es raiz".format(x0))
else:
    x1 = x0 + delta
    contador = 1
    fx1 = f.evalf(subs={x:x1})

    while fx0 * fx1 > 0 and contador <= n_iter:
        x0 = x1
        fx0 = fx1
        x1 = x0 + delta
        fx1 = f.evalf(subs={x:x1})
        contador += 1

    if fx1 == 0:
        print("{0} es raiz".format(x0))
    elif (fx0 * fx1) < 0:
        print("Hay una raiz entre {0} y {1}".format(x0, x1))
    else:
        print("Fracaso en {0} iteraciones".format(n_iter))


